### PR TITLE
Refactor some pieces of code before implementing local drop & truncate execution

### PR DESCRIPTION
**Pull request by onurctirtir**
This pr aims to:
* Fix some typos around the local execution
* Some cleanup in `local_executor.h`
* Refactor `DropShards` function before introducing local execution of `DROP` commands

### CODE REVIEW

File: src/backend/distributed/master/master_delete_protocol.c [hunk 1]
```diff
@@ -359,99 +365,169 @@ static int
...
+										 char *dropShardPlacementCommand)
+{
+	Assert(shardPlacement != NULL);
+	Assert(shardRelationName != NULL);
+	Assert(dropShardPlacementCommand != NULL);
 
-				continue;
-			}
+	uint32 dropShardConnectionFlags = FOR_DDL;
+	MultiConnection *connection = GetPlacementConnection(dropShardConnectionFlags,
```
Comment (onderkalaci) [inline src/backend/distributed/master/master_delete_protocol.c]: we typically always use `connectionFlags`, so `dropShard ` prefix seems a little unnecessary

File: src/backend/distributed/master/master_delete_protocol.c [hunk 2]
```diff
@@ -359,99 +365,169 @@ static int
...
-			else if (storageType == SHARD_STORAGE_COLUMNAR ||
-					 storageType == SHARD_STORAGE_FOREIGN)
+			else
 			{
-				appendStringInfo(workerDropQuery, DROP_FOREIGN_TABLE_COMMAND,
-								 quotedShardName);
+				char storageType = shardInterval->storageType;
+
+				/* build command to drop the shard */
+				const char *dropShardPlacementCommand = CreateDropShardPlacementCommand(
```
Comment (onderkalaci) [inline src/backend/distributed/master/master_delete_protocol.c]: minor: more typical indentation
```C
const char *dropShardPlacementCommand = 
    CreateDropShardPlacementCommand(schemaName, shardRelationName, 
                                    storageType);
```

File: src/backend/distributed/master/master_delete_protocol.c [hunk 3]
```diff
@@ -359,99 +365,169 @@ static int
...
+					schemaName, shardRelationName, storageType);
+
+				/*
```
Comment (onderkalaci) [inline src/backend/distributed/master/master_delete_protocol.c]: Minor: I think the function name pretty clear, so the comment seems a little unnecessary, consider removing it

File: src/backend/distributed/executor/local_executor.c [hunk 4]
```diff
@@ -228,6 +229,25 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 }
 
 
+/*
+ * LogLocalCommand logs commands executed locally on this node. Although we're
```
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/backend/distributed/executor/local_executor.c]: just to keep it close to where it is used

File: src/include/distributed/local_executor.h [hunk 5]
```diff
@@ -20,16 +20,14 @@ extern bool LogLocalCommands;
 extern bool TransactionAccessedLocalPlacement;
 extern bool TransactionConnectedToLocalGroup;
 
+/* extern function declarations */
 extern uint64 ExecuteLocalTaskList(CitusScanState *scanState, List *taskList);
 extern void ExtractLocalAndRemoteTasks(bool readOnlyPlan, List *taskList,
 									   List **localTaskList, List **remoteTaskList);
 extern bool ShouldExecuteTasksLocally(List *taskList);
-extern void ErrorIfTransactionAccessedPlacementsLocally(void);
```
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/include/distributed/local_executor.h]: maybe there can be some automated tools but I just found them manually

File: src/backend/distributed/master/master_delete_protocol.c [hunk 6]
```diff
@@ -359,99 +365,169 @@ static int
...
+				ExecuteDropShardPlacementCommandRemotely(shardPlacement,
+														 shardRelationName,
+														 dropShardPlacementCommand);
 			}
 
 			/*
-			 * The active DROP SCHEMA/DATABASE ... CASCADE will drop the shard, if we
-			 * try to drop it over another connection, we will get into a distributed
-			 * deadlock.
+			 * Regardless of that we dropped the placement or not, delete the
```
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/backend/distributed/master/master_delete_protocol.c]: removing makes more sense to me, done

File: src/backend/distributed/master/master_delete_protocol.c (hunk 1 above)
Comment (onurctirtir) [reply src/backend/distributed/master/master_delete_protocol.c]: makes sense, reverting

File: src/backend/distributed/master/master_delete_protocol.c [hunk 7]
```diff
@@ -359,99 +365,169 @@ static int
...
-								  errdetail("Marking this shard placement for "
-											"deletion")));
 
-				UpdateShardPlacementState(placementId, SHARD_STATE_TO_DELETE);
+/*
+ * ExecuteDropShardPlacementCommandRemotely executes the given DROP shard command
+ * via remote critical connection.
+ */
+static void
+ExecuteDropShardPlacementCommandRemotely(ShardPlacement *shardPlacement, const
```
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/backend/distributed/master/master_delete_protocol.c]: yeap, I guess that was done by reindent. fixing it

**Pull request merged**